practice/node/exercises-for-programmers/chapter-6/GuesstheNumberGame/script.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+const fs = require('fs');
+const readlineSync = require('readline-sync');
+
+
+
+const difficulty = readlineSync.question('Level of Difficulty : ');
+let numOfGuesses = 0 
+let correctGuess = false;
+
+switch (difficulty){
+    case "hard":
+        let hardRandomNum= Math.floor(Math.random() * Math.floor(1000));
+        
+    while (correctGuess=== false){
+        const guess = readlineSync.question('Guess the Number : ');
+        if(guess > hardRandomNum){
+            console.log("Too High")
+            numOfGuesses ++
+        } else if(guess < hardRandomNum){
+            console.log("Too Low")
+            numOfGuesses ++
+        } else {
+            console.log("You've got it")
+            console.log("Number of Guesses  : " + numOfGuesses)
+            correctGuess = true
+        }
+
+    }
```

Comment:
👏 Nice! It looks like the only the only thing that changes between your `while` loops is the value for the random number. Consider extracting the contents of the `while` loop to a separate function that accepts the random number as an argument, then calling this function for each case, passing the random number as an argument. You could move your declaration of your `correctGuess` variable inside of that function as well.

Reply:
I get exactly what you mean. Great way to think of doing this. thank you for your feedback.

Reply:
@colbyallenc Don't forget the part that randomizes the number. Thinking about this some more, maybe the function argument should be the max value (10, 100, 1000), and inside of your function you could declare the random number based on that value:
```
const randomNum = Math.floor(Math.random() * Math.floor(maxValue))
```

Reply:
I was so quick and excited to solve it that I forgot to randomize the number. I made another pull that shows the randomization